tests/e2e/question.pageObject.js

```diff
@@ -27,76 +30,145 @@ var QuestionPage = function() {
   const pageUrl = '/client/question.html';
 
   /**
-   * DOM element where the code is input
+   * TIE wrapper element.
+   *
+   * @type {webdriver.WebElement}
+   */
+  var tieWrapperElement = element(by.css('.protractor-test-tie-wrapper'));
+
+  /**
+   * Question UI element.
+   *
+   * @type {webdriver.WebElement}
+   */
+  var questionUiElement = element(by.css('.protractor-test-question-ui'));
+
+  /**
+   * Coding UI element.
+   *
+   * @type {webdriver.WebElement}
+   */
+  var codingUiElement = element(by.css('.protractor-test-coding-ui'));
+
+  /**
+   * Reset Code button.
    *
-   * @type {DOMElement}
+   * @type {webdriver.WebElement}
    */
-  var codeInput = element(by.css('.protractor-test-code-input'));
+  var resetCodeButton = element(by.css('.protractor-test-reset-code-button'));
 
   /**
-   * Set of all of the feedback paragraphs rendered in the DOM.
+   * Run Code button.
    *
-   * @type {Array}
+   * @type {webdriver.WebElement}
+   */
+  var runCodeButton = element(by.css('.protractor-test-run-code-button'));
+
+  /**
+   * Set of all of the feedback text paragraphs rendered in the DOM.
+   *
+   * @type {Array.<webdriver.WebElement>}
    */
   var feedbackParagraphs =
       element.all(by.css('.protractor-test-feedback-paragraph'));
 
   /**
-   * DOM element where the reset code button is.
+   * Reset Feedback button.
+   *
+   * @type {webdriver.WebElement}
+   */
+  var resetFeedbackButton =
+      element(by.css('.protractor-test-reset-feedback-button'));
+
+  /**
+   * Theme selector.
+   *
+   * @type {webdriver.WebElement}
+   */
+  var themeSelector = element(by.css('.protractor-test-theme-select'));
+
+  /**
+   * Python primer link.
    *
-   * @type {DOMElement}
+   * @type {webdriver.WebElement}
    */
-  var resetCodeBtn = element(by.css('.protractor-test-reset-code-btn'));
+  var pythonPrimerLink = element(by.css('.protractor-test-python-primer-link'));
 
   /**
-   * DOM element where the run code button is.
+   * TIE About link.
    *
-   * @type {DOMElement}
+   * @type {webdriver.WebElement}
    */
-  var runCodeBtn = element(by.css('.protractor-test-run-code-btn'));
+  var aboutLink = element(by.css('.protractor-test-about-link'));
+
+  /**
+   * TIE Privacy link.
+   *
+   * @type {webdriver.WebElement}
+   */
+  var privacyLink = element(by.css('.protractor-test-privacy-link'));
+
+  /**
+   * TIE Leave Feedback link.
+   *
+   * @type {webdriver.WebElement}
+   */
+  var feedbackLink = element(by.css('.protractor-test-feedback-link'));
 
   /**
    * Retrieves the TIE question page.
    *
-   * @param {string} questionId ID of the question to open.
+   * @param {string} questionId ID of the question to load.
+   * @param {boolean} setupPage Whether or not to setup the page after load.
```

Comment:
Hm, the name sounds like it represents a page object. Maybe rename to runSetupPageFunction or shouldSetupPageFunctionBeRun or similar?

(Basically, in this codebase, this change would cause setupPage to be used as both a function name and a boolean variable name. I think we should make the boolean variable more obviously boolean.)

Reply:
Actually, thinking about it more -- why do we even need this option? Are there cases when we shouldn't be setting up the page after load? It would be better if branching isn't needed.

Reply:
There is one specific case in the server version where we need to bypass page setup in order to test the presence of the privacy modal, changing the name of the argument to runSetupPageFunction